sweep/txgenerator.go

```diff
@@ -339,6 +339,8 @@ func getWeightEstimate(inputs []input.Input, outputs []*wire.TxOut,
 	// trimmed.
```

Comment:
would be cool if we could have some way to make sure that this if-else chain (which could be a switch) checks the same cases that are checked in [CraftSweepAllTx](https://github.com/lightningnetwork/lnd/blob/fec8fd9c63dc672dc3ea1b9be47bd1d6c4a0a54d/sweep/walletsweep.go#L246) so that this bug is prevented in future if any new script types are added.

Reply:
I think logic unification would be good, but not sure how to do it here. The cases here need to check more cases than the `CraftSweepAllTx` cases (since those are only wallet UTXOs -- fewer possible script types). Also the `CraftSweepAllTx` case needs to check for taproot pubkey rather than the general v1 witness script